orchagent/fdborch.cpp

```diff
@@ -36,6 +37,73 @@ FdbOrch::FdbOrch(DBConnector *db, string tableName, PortsOrch *port) :
     Orch::addExecutor(fdbNotifier);
 }
 
+bool FdbOrch::bake()
+{
+    Orch::bake();
+
+    auto consumer = dynamic_cast<Consumer *>(getExecutor(APP_FDB_TABLE_NAME));
+    if (consumer == NULL)
+    {
+        SWSS_LOG_ERROR("No consumer %s in Orch", APP_FDB_TABLE_NAME);
+        return false;
+    }
+
+    size_t refilled = consumer->refillToSync(&m_fdbStateTable);
+    SWSS_LOG_NOTICE("Add warm input FDB State: %s, %zd", APP_FDB_TABLE_NAME, refilled);
+    return true;
+}
+
+bool FdbOrch::storeFdbEntryState(const FdbUpdate& update)
+{
+    const FdbEntry& entry = update.entry;
+    const Port& port = update.port;
+    sai_vlan_id_t vlan_id = port.m_port_vlan_id;
+    const MacAddress& mac = entry.mac;
+    string portName = port.m_alias;
+
+    // ref: https://github.com/Azure/sonic-swss/blob/master/doc/swss-schema.md#fdb_table
+    string key = "Vlan" + to_string(vlan_id) + ":" + mac.to_string();
```

Comment:
>: [](start = 48, length = 1)

should this be '|'

Reply:
Currently there is a trick. All entries stored in StateDB FDB_TABLE will be fed into ApplDB during warm starting. So I keep them the same pattern.

---
In reply to: [219357862](https://github.com/Azure/sonic-swss/pull/615#discussion_r219357862) [](ancestors = 219357862)

Reply:
I still feel it is more convenient to save the data in appDB using a new table like ASIC_FDB_TABLE, so to eliminate all the exception handling (stateDB save/restore during system warm reboot, and the special ":" separator in stateDB).